Fix: sample builder test always returns data

robin-raymond opened the pull request:
Wrap around issue caused backup of packets in sample builder.

#### Description

#### Reference issue
Fixes #1870

Sean-Der commented:
Thanks for grabbing this @robin-raymond !

jech commented:
So you're committing my code under your name, now?  @robin-raymond, that's not how it's done in polite society.

robin-raymond commented:
@jech I'm not going to engage in this... simple solution, work with @Sean-Der  to revert the change and attach the test + fix to your name since it's important to you to get your name attached to a unit test. I would have thought you to be happy the issue was resolved. BTW, your welcome for fixing a one line bug that caused a fork for you. I had asked for a unit test 24 days ago here: https://github.com/pion/webrtc/issues/1839 and when you finally gave one a long time later it was unclear what the test was about and looked like you were attempting to introduce new behaviour as you had not mentioned the issue was only during the wrap around point (very important clue to note) and not a generalized usage scenario. I had only realized a potential issue existed which someone else posted about. Clearly there should have been a unit test about the wrap around point from day one on the original code but alas sometimes things get missed and that's understandable. Your test was the most expedient to demonstrating the fix addresses your concerns and allowed me to demonstrate a fix a bug that was hurting you and causing a fork. I spent my time to help you and the project as a whole but instead of being happy to have a fix you malign as demonstrated above. I really don't care and have never cared about that kind of thing. I have enough credits to my name on many projects, if I even cared (which I don't). I have always been interested in ensuring the welfare of any project I have ever worked on and not personal kudos.
